sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -655,8 +655,17 @@ class Dataset[T] private[sql](
    * @group basic
    * @since 2.4.0
    */
-  def isEmpty: Boolean = withAction("isEmpty", select().limit(1).queryExecution) { plan =>
-    plan.executeTake(1).isEmpty
+  def isEmpty: Boolean = {
+    val newDs = logicalPlan match {
+      case c: CommandResult =>
```

Comment:
I wonder if we can deduplicate those.

Reply:
maybe add a new `private def withCommandResultOptimized: Dataset[T]` to reuse code?

Reply:
> maybe add a new `private def withCommandResultOptimized: Dataset[T]` to reuse code?

SGTM, I will add it later